ui/app/components/ui/TableItems.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from "react";
+import { Link } from "react-router";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipTrigger,
+  TooltipProvider,
+} from "./tooltip";
+import { formatDate } from "~/utils/date";
+
+interface TableItemIdProps {
+  id: string;
+  link?: string;
+}
+
+function TableItemId({ id, link }: TableItemIdProps) {
+  const content = (
+    <span className="font-mono whitespace-nowrap">
+      {id.length > 8 ? `…${id.slice(-8)}` : id}
+    </span>
+  );
+
+  return (
+    <TooltipProvider delayDuration={400}>
+      <Tooltip>
+        <TooltipTrigger asChild>
+          {link ? (
+            <Link
+              to={link}
+              className="block no-underline transition-colors duration-300 hover:text-gray-500"
```

Comment:
```suggestion
              aria-label={id}
              className="block no-underline transition-colors duration-300 hover:text-gray-500"
```
Useful to label the link since the actual contents may be truncated.